client/src/components/campaignCreatorComponents/CampaignCreator.tsx

```diff
@@ -36,8 +43,81 @@ const CampaignCreator = ({ campaignSettings, setCampaignSettings }: CampaignCrea
         navigate('/twilight-library/dashboard');
     };
 
+    const addToTimeline = (
+        timeline: Record<number, string[]>,
+        nodeKey: keyof NodePillarLists,
+        typeKey: keyof TimelineOptionList,
+    ) => {
+        if (!selectedCampaign.flexible_nemesis_encounters && typeKey === 'nemesis') {
+            // Handle strict Nemesis timeline
+            const nemesis_tier: string = nodeKey.slice(-1);
+            selectedCampaign[nodeKey].forEach((selection: string) => {
+                [1, 2, 3].forEach((level: number) => {
+                    const fight_str: string = `nn${nemesis_tier}_lvl${level}_fight_year`;
+                    const fight_key: keyof NemesisFightYearLists = fight_str as keyof NemesisFightYearLists;
+                    const fight_year: number | null = selectedCampaign[fight_key];
+                    if (fight_year !== null) {
+                        timeline[fight_year].push(`NE - ${selection} lvl ${level}`);
+                    }
+                });
+            });
+            return;
+        }
+
+        selectedCampaign[nodeKey].forEach((selection: string) => {
+            const query: Record<string, Record<string, string[]>> = campaignOptionsData.timeline[typeKey];
+            if (query.hasOwnProperty(selection)) {
+                Object.keys(query[selection]).forEach((yearKey: string) => {
+                    query[selection][Number(yearKey)].forEach((yearData: string) => {
+                        timeline[Number(yearKey)].push(yearData);
+                    });
+                });
+            }
+        });
+    };
+
     const handleCreateTimeline = (event: React.FormEvent<HTMLFormElement>) => {
         event.preventDefault();
+        if (settlementName.length < 2) {
+            // TODO: Add Toast popup with warning
+            return;
+        }
+        // Populate Timeline
+        const maximumYears: number = 40;
+        const timeline: Record<number, string[]> = Array.from({ length: maximumYears }, (_, index) => index + 1).reduce(
+            (obj, key) => ({ ...obj, [key]: [] }),
+            {},
+        );
```

Comment:
Clever, but I don't know if there's actually much use case for instantiating empty data

Reply:
it simplifies timeline data use once saved to the database. I'm going leave it.